hashicorp/boundary-ui
# chore: 🤖 Create `select` field render for the `Form::Field::ListWrapper`

✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12271
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-11719
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12271)

## Description
The `Form::Field::ListWrapper` [component ](https://github.com/hashicorp/boundary-ui/tree/main/ui/admin/app/components/form/field/list-wrapper)was created to simplify the process of adding and removing input fields in a form, which otherwise involved a lot of repetitive and boilerplate code when using the `Hds::Table` component. The component provides the `KeyValue`, `TextInput`, and `TextArea` components as options to the user, but at present, it does not support the `SelectField` component.

- Create `select` field  yield that allows users to have a list of `select` fields that they can add and remove.
- Test using the form for `OIDC` auth-method



## Screenshots (if appropriate):

<img width="1673" alt="Screenshot 2024-01-25 at 11 09 41" src="https://github.com/hashicorp/boundary-ui/assets/24277002/8c8049da-2deb-4f4c-8110-461dd5f68520">

<img width="1673" alt="Screenshot 2024-01-25 at 11 10 00" src="https://github.com/hashicorp/boundary-ui/assets/24277002/d2770324-8c21-4834-a576-80cf000d31a5">

## How to Test
- Click on Vercel link for Admin UI or pull down PR and run it locally using `yarn start`
- Click on "Auth Method" in the sidebar navigation.
- Click on the "New" button in the top right hand corner of page and select OIDC.
- Ensure in the "Signing Algorithms" section you are able to select the different algorithms and add them to the table.

## Checklist:


~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have added steps to reproduce and test for bug fixes in the description~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
